js/main.ts

```diff
@@ -2,25 +2,22 @@
 // tslint:disable-next-line:no-reference
 /// <reference path="./plugins.d.ts" />
 
-import "./globals";
+import { window } from "./globals";
 
 import { log } from "./util";
 import * as os from "./os";
 import { libdeno } from "./libdeno";
 import { args } from "./deno";
 import { replLoop } from "./repl";
 
-// builtin modules
-import * as deno from "./deno";
-
 // TODO(kitsonk) remove with `--types` below
 import libDts from "gen/lib/lib.deno_runtime.d.ts!string";
 
 /* tslint:disable-next-line:no-default-export */
 export default function denoMain() {
   const startResMsg = os.start();
 
-  libdeno.builtinModules["deno"] = deno;
+  libdeno.builtinModules["deno"] = window.deno;
```

Comment:
Backwards compatibility, we still can do this:
```ts
import * as deno from "deno";
```

Reply:
But `window.deno` is preferred.

Reply:
Please add a TODO to remove the builtinModules at some point. It's a huge amount of complexity that, if possible, would be nice to avoid.